### DISCUSSION TITLE
Issue with undefined in API call using useSWR from second collection in mongoDB

### DISCUSSION THREAD

**Discussion by migueljcpinto** [neuefische/questions#677] [Web] [answered]
I'm developing my project with mongoDB and have encountered an issue when trying to fetch the work dates for a specific nurse using the useSWR library. When making a call to my API, the nurse's ID is being passed as undefined, resulting in a 500 error.

In my component, I'm attempting to get the nurse's ID as follows:

```
  const router = useRouter();
  const { id } = router.query; //Nurse id
  const { data: nurseData, isLoading, mutate } = useSWR(id ? `/api/nurses/${id}` : null);
  const { data: workDatesData } = useSWR(id ? `/api/work-dates/${id}` : null, fetcher);
```

In the console, I'm seeing the following error:

`GET http://localhost:3000/api/nurses/undefined 500 (Internal Server Error) fetcher.js:2     
  GET http://localhost:3000/api/work-dates/undefined 500 (Internal Server Error) fetcher.js:2
`

I've noticed that the value of router.query.id might not be immediately available, which could be causing the undefined.
The issue arises when I navigate to the page making this API call.
My fetcher utility seems to be working correctly for other routes (like for Nurse Team), but I'm unsure if it's contributing to this problem.

Perhaps I'm accessing router.query.id too early, before it has a value, or is it a routing problem because I restructured all the code in new files and folders.

That is my [Code](https://github.com/MiketheBite/Capstone-project/tree/userStory4lastVersion).

Any help or suggestions to solve this issue would be greatly appreciated!

**Comment by F-Kirchhoff** [comment, marked as answer]
it seems that you don't pass the nurseData.id correctly in the link to the schedule page (components/NurseProfile). You wrote
```js
href=`/nurses/${nurseData.id}/schedule`
```
but i guess it should be 
```js
href=`/nurses/${nurseData._id}/schedule`
```
if you are using the mongoDB id.

**Reply by migueljcpinto** [reply]
Great!!! I think we arrived at this solution at the same time, which I was already changing the question post when you answered. 😁 But, I still don't receive the information in the MongoDB collection. Can you check this too? Or will I have to open a new discussion post on the channel?

**Reply by F-Kirchhoff** [reply]
Maybe I come to your breakout room and we check it together, without a database the bug hunt is quite difficult.

**Reply by migueljcpinto** [reply]
Thanks a lot. I figured it out. 🤙🍻
